zcash_primitives/src/merkle_tree.rs

```diff
@@ -117,7 +121,35 @@ impl<Node> CommitmentTree<Node> {
         }
     }
 
-    pub fn to_frontier(&self) -> bridgetree::Frontier<Node, SAPLING_COMMITMENT_TREE_DEPTH_U8>
+    pub fn from_frontier<const DEPTH: u8>(frontier: &bridgetree::Frontier<Node, DEPTH>) -> Self
+    where
+        Node: Clone,
+    {
+        frontier.value().map_or_else(Self::empty, |f| {
+            let (left, right) = match f.leaf() {
+                Leaf::Left(v) => (Some(v.clone()), None),
+                Leaf::Right(l, r) => (Some(l.clone()), Some(r.clone())),
+            };
+            let mut ommers_iter = f.ommers().iter().cloned();
+            let upos: usize = f.position().into();
+            Self {
+                left,
+                right,
+                parents: (1..DEPTH)
+                    .into_iter()
+                    .map(|i| {
+                        if upos & (1 << i) == 0 {
+                            None
+                        } else {
+                            ommers_iter.next()
+                        }
```

Comment:
I confirmed that this lines up with how `CommitmentTree::size` is implemented:

https://github.com/zcash/librustzcash/blob/43c18d000fcbe45363b2d53585d5102841eff99e/zcash_primitives/src/merkle_tree.rs#L156-L160